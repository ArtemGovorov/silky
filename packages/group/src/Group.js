// @flow
import React from 'react';

type props = {
  children: any,
  className: string,
  position: {
    left: number,
    top: number,
  },
};

export default function Group({
  children,
  className,
  position: { left = 0, top = 0 } = {},
  ...restProps
}: props) {
  return (
    <g
      className={`silky-group ${className}`}
      transform={`translate(${left}, ${top})`}
      {...restProps}
    >
      {children}
    </g>
  );
}
