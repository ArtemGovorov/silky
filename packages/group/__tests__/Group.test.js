import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Group } from '../src';

afterEach(cleanup);

test('Group', () => {
  const { container } = render(
    <svg>
      <Group>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.innerHTML).toBe('foo');
});

test('Group with custom class', () => {
  const { container } = render(
    <svg>
      <Group className="foo">foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('silky-group')).toBe(true);
  expect(group.classList.contains('foo')).toBe(true);
  expect(group.innerHTML).toBe('foo');
});

test('Group width custom position', () => {
  const { container } = render(
    <svg>
      <Group position={{ left: 10, top: 10 }}>foo</Group>
    </svg>
  );

  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.getAttribute('transform')).toEqual('translate(10, 10)');
  expect(group.innerHTML).toBe('foo');
});
