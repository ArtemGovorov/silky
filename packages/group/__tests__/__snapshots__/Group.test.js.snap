// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group 1`] = `
<g
  class="silky-group undefined"
  transform="translate(0, 0)"
>
  foo
</g>
`;

exports[`Group width custom position 1`] = `
<g
  class="silky-group undefined"
  transform="translate(10, 10)"
>
  foo
</g>
`;

exports[`Group with custom class 1`] = `
<g
  class="silky-group foo"
  transform="translate(0, 0)"
>
  foo
</g>
`;
